met-web/src/components/survey/create/CloneOptions.tsx

```diff
@@ -0,0 +1,232 @@
+            setAvailableSurveys(fetchedSurveys.items);
+            setLoadingSurveys(false);
+        } catch (error) {
+            dispatch(openNotification({ severity: 'error', text: 'Error occurred while fetching available surveys' }));
+        }
+    };
+
+    const handleFetchEngagements = async (
+        page: number,
+        size: number,
+        sort_order: 'asc' | 'desc' | undefined,
+        search_text?: string,
+    ) => {
+        try {
+            const fetchedEngagements = await getEngagements({
+                page: page,
+                size: size,
+                sort_order: sort_order,
+                search_text: search_text,
+            });
+            console.log('FETCH!');
+            console.log(fetchedEngagements);
+            setAvailableEngagements(
+                fetchedEngagements.items.filter((engagement, index) => engagement.surveys.length !== 0),
+            );
+            setLoadingEngagements(false);
+        } catch (error) {
+            dispatch(
+                openNotification({ severity: 'error', text: 'Error occurred while fetching available engagements' }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        if (!availableSurveys) {
+            handleFetchSurveys(1, 10, 'asc', '');
+        }
+    }, [availableSurveys]);
+
+    useEffect(() => {
+        if (!availableEngagements) {
+            handleFetchEngagements(1, 10, 'asc', '');
+        }
+    }, [availableEngagements]);
+
+    const handleSave = async () => {
+        if (!selectedSurvey) {
+            if (selectedEngagement) return;
+            dispatch(openNotification({ severity: 'error', text: 'Please select a survey first' }));
+            return;
+        }
+        setIsSaving(true);
+        try {
+            const createdSurvey = await postSurvey({
+                name: surveyForm.name,
+                form_json: selectedEngagement ? selectedEngagement.surveys[0].form_json : selectedSurvey.form_json,
+            });
+
+            dispatch(
+                openNotification({
+                    severity: 'success',
+                    text: 'Survey created, please proceed to building it',
+                }),
+            );
+            navigate(`/surveys/${createdSurvey.id}/build`);
+        } catch (error) {
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while attempting to save survey',
+                }),
+            );
+            setIsSaving(false);
+        }
+    };
+
+    return (
+        <Grid container direction="row" alignItems="flex-start" justifyContent="flex-start" item xs={12} spacing={2}>
+            <Grid item xs={6}>
+                <MetLabel sx={{ marginBottom: '2px', display: 'flex' }}>
+                    Select Engagement <Typography sx={{ ml: 1, color: '#ACA9A9' }}>(optional)</Typography>
+                </MetLabel>
+
+                <Autocomplete
```

Comment:
select engagement is optional, should be used to filter the surveys select when used. If surveys are 100 and select engagement is used to select an engagement then of those 100 we return only the surveys that have engagement_id equal to the selectedEngagement id in our case it will always be one survey until later when we might have multiple surveys for one engagement, or for one project.